sw/device/tests/crypto/ecdsa_p256_verify_set_testvectors.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python3
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+import argparse
+import os
+import sys
+
+import hjson
+from mako.template import Template
+from pathlib import Path
+
+'''
+Read in a JSON test vector file, convert the test vector to C constants, and
+generate a header file with these test vectors.
+'''
+
+# Number of 32-bit words in a coordinate or scalar
+P256_NUMWORDS = int(256 / 32)
+
+# Template file name
+TEMPLATE = 'ecdsa_p256_verify_testvectors.h.tpl'
+
+# Default output file name
+DEFAULT_OUTFILE = 'ecdsa_p256_verify_testvectors.h'
+
+
+def ecdsa_p256_int_to_hexwords(x):
+    '''Convert a 256-bit integer to a list of 32-bit integers (little-endian).'''
+    out = []
+    for _ in range(P256_NUMWORDS):
+        out.append(x & ((1 << 32) - 1))
+        x >>= 32
+    if x != 0:
+        raise ValueError('Integer out of range!')
+    return ['{:#010x}'.format(w) for w in out]
+
+
+def main() -> int:
+    parser = argparse.ArgumentParser()
+    parser.add_argument('hjsonfile',
+                        metavar='FILE',
+                        type=argparse.FileType('r'),
+                        help='Read test vectors from this HJSON file.')
+    out_default = open(
+        os.path.join(os.path.dirname(__file__), DEFAULT_OUTFILE), 'w')
+    parser.add_argument('headerfile',
```

Comment:
Good to know -- I think I will leave this one for now because the other test files (e.g. RSA and sigverify) have the same structure, and do a follow-up to change them all at once.

Reply:
There is probably a safe way to do this with `argparse.FileType`, something similar to this should work:

```python
with args.hjsonfile as f:
  testvecs = hjson.load(f)
```

Reply:
I've just read https://bugs.python.org/issue13824 carefully, which is about exactly this problem. And it looks like that's a perfect solution (and doesn't kill stdin/stdout, which is what I thought it would do). Neat!

Reply:
Created https://github.com/lowRISC/opentitan/issues/11106 to track this and a couple other comments from this PR that should be propagated to RSA/sigverify scripts.